### PR TITLE
[MB-7624] update SFTP reader to user GEX env vars

### PR DESCRIPTION
**Pull request by jacquelineIO**
## Description

Update the process EDIs functions to use GEX SFTP instead of using Syncada SFTP. I kept the Syncada SFTP file intact since we still have all of the chamber envs to go with it. I will create a new PR to remove the Syncada ENVs and the code files that go with it.

## Reviewer Notes

**This code cannot be tested locally. I guess running to see that you fail locally is the only thing that you can do. This will have to be tested in staging.**

All of the GEX envs should be present in chamber. 
```sh
aws-vault exec transcom-gov-milmove-stg -- chamber env app-stg  | grep -i gex_sftp
```

Just check that there are values for all of the following:
gex_sftp_host
gex_sftp_host_key
gex_sftp_ip_address
gex_sftp_password
gex_sftp_port
gex_sftp_user_id
gex_sftp_997_pickup_directory
gex_sftp_824_pickup_directory

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
echo "Code goes here"
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: pkg/services/invoice/ghc_payment_request_invoice_generator_test.go [hunk 1]
```diff
@@ -853,43 +853,7 @@ func (suite *GHCInvoiceSuite) TestTruncateStrFunc() {
...
-		basicPaymentServiceItemParams,
-		assertions,
-	)
-	testdatagen.MakePaymentServiceItemWithParams(
-		suite.DB(),
-		models.ReServiceCodeCS,
-		basicPaymentServiceItemParams,
-		assertions,
-	)
+	paymentRequest := testdatagen.MakePaymentRequest(suite.DB(), testdatagen.Assertions{})
```
Comment (jacquelineIO) [inline pkg/services/invoice/ghc_payment_request_invoice_generator_test.go:856]: @monfresh  UT was updated here

File: pkg/cli/gex_sftp.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	// GEXSFTPUserIDFlag is the ENV var for the GEX SFTP user ID
+	GEXSFTPUserIDFlag string = "gex-sftp-user-id"
+	// GEXSFTPIPAddressFlag is the ENV var for the GEX SFTP IP address
+	GEXSFTPIPAddressFlag string = "gex-sftp-ip-address"
+	// GEXSFTPPasswordFlag is the ENV var for the GEX SFTP password
+	GEXSFTPPasswordFlag string = "gex-sftp-password" // #nosec G101
+	// GEXSFTPHostKeyFlag is the ENV var for the GEX SFTP host key
+	GEXSFTPHostKeyFlag string = "gex-sftp-host-key"
+	// GEXSFTP997PickupDirectory is the ENV var for the directory where GEX delivers responses
+	GEXSFTP997PickupDirectory string = "gex_sftp_997_pickup_directory"
```
Comment (mboren) [inline pkg/cli/gex_sftp.go:26]: nit: Is the switch from dashes to underscores here intentional?

File: pkg/cli/gex_sftp.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+// TODO: Figure out what to do in terms of static analysis on the false positive for G101
+const (
+	// GEXSFTPPortFlag is the ENV var for the GEX SFTP port
+	GEXSFTPPortFlag string = "gex-sftp-port"
+	// GEXSFTPUserIDFlag is the ENV var for the GEX SFTP user ID
+	GEXSFTPUserIDFlag string = "gex-sftp-user-id"
+	// GEXSFTPIPAddressFlag is the ENV var for the GEX SFTP IP address
+	GEXSFTPIPAddressFlag string = "gex-sftp-ip-address"
+	// GEXSFTPPasswordFlag is the ENV var for the GEX SFTP password
+	GEXSFTPPasswordFlag string = "gex-sftp-password" // #nosec G101
```
Comment (gidjin) [inline pkg/cli/gex_sftp.go:22]: Link to the story here that notes we are going to be switching away from password to key for authentication. Also should note for the other reviewers this PR is required to verify our connection which is required before we can switch to the private key. See https://dp3.atlassian.net/browse/MB-7728

**Review by mboren — APPROVED** [review]
LGTM except for one potential typo, and I'm not sure if we have to add more annotations for for static analysis before merging. Looks like #6238 finally got approval today, so maybe that helps?

File: cmd/fetch-from-syncada-via-sftp/main.go [hunk 4]
```diff
@@ -21,8 +21,10 @@ import (
 const (
 	// LastReadTimeFlag is the ENV var for the last read time
 	LastReadTimeFlag string = "last-read-time"
-	// DirectoryFlag is the ENV var for the directory
-	DirectoryFlag string = "directory"
+	// Directory997Flag is the ENV var for the directory
+	Directory997Flag string = "directory"
+	// Directory824Flag is the ENV var for the directory
+	Directory824Flag string = "directory"
```
Comment (gidjin) [inline cmd/fetch-from-syncada-via-sftp/main.go:27]: I wonder if we should get rid of this cli tool since we can call the process-edi task on the cli too. Definitely not part of this PR, imo.

File: pkg/services/invoice/syncada_sftp_reader.go [hunk 5]
```diff
@@ -56,22 +55,9 @@ func (s *syncadaReaderSFTPSession) FetchAndProcessSyncadaFiles(syncadaPath strin
...
-	// dirs, then we won't need to inspect the file to determine the type.
-	var checkEDIType *regexp.Regexp
-	if processor.EDIType() == models.EDIType997 {
-		checkEDIType, err = regexp.Compile(`ST\*997\*\d*`)
-	} else if processor.EDIType() == models.EDIType824 {
-		checkEDIType, err = regexp.Compile(`ST\*824\*\d*`)
-	}
+	fileList, err := s.client.ReadDir(pickupPath)
 	if err != nil {
+		s.logger.Error("Error reading SFTP directory", zap.String("directory", pickupPath))
```
Comment (gidjin) [inline pkg/services/invoice/syncada_sftp_reader.go:60]: Do we have a successful log message on what directory we're reading? If not might be good to add a message noting we are reading from `pickupPath` or something so we know it's good in the happy path as well.

**Review by gidjin — APPROVED** [review]
Changes look good. Locally it hangs connecting to sftp as expected. Left a couple comments.

File: pkg/cli/gex_sftp.go (hunk 2 above)
Comment (jacquelineIO) [reply pkg/cli/gex_sftp.go:26]: nope! this was a mistake good catch

File: pkg/services/invoice/syncada_sftp_reader.go (hunk 5 above)
Comment (jacquelineIO) [reply pkg/services/invoice/syncada_sftp_reader.go:60]: I will check to add the messages.

Comment (jacquelineIO) [reply pkg/services/invoice/syncada_sftp_reader.go:60]: success logging is present. pinged you in Slack to see if more should be added

File: pkg/services/invoice/ghc_payment_request_invoice_generator_test.go (hunk 1 above)
Comment (monfresh) [reply pkg/services/invoice/ghc_payment_request_invoice_generator_test.go:856]: @jacquelineIO Thanks, although what I meant was that we shouldn't be creating any payment requests in the DB at all. We can pass in an empty model `models.PaymentRequest{}` into the Generate function.
